core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/aggregation/MinAggregator.java

```diff
@@ -0,0 +1,100 @@
+import static com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils.getStringValue;
+import static com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils.integerValue;
+import static com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils.longValue;
+import static com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType.DOUBLE;
+import static com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType.FLOAT;
+import static com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType.INTEGER;
+import static com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType.LONG;
+import static com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType.SHORT;
+import static com.amazon.opendistroforelasticsearch.sql.utils.ExpressionUtils.format;
+
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprNullValue;
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprValue;
+import com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType;
+import com.amazon.opendistroforelasticsearch.sql.exception.ExpressionEvaluationException;
+import com.amazon.opendistroforelasticsearch.sql.expression.Expression;
+import com.amazon.opendistroforelasticsearch.sql.expression.function.BuiltinFunctionName;
+import com.amazon.opendistroforelasticsearch.sql.storage.bindingtuple.BindingTuple;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The minimum aggregator aggregate the value evaluated by the expression.
+ * If the expression evaluated result is NULL or MISSING, then the result is NULL.
+ */
+public class MinAggregator extends Aggregator<MinAggregator.MinState> {
+
+  public MinAggregator(List<Expression> arguments, ExprCoreType returnType) {
+    super(BuiltinFunctionName.MIN.getName(), arguments, returnType);
+  }
+
+
+  @Override
+  public MinState create() {
+    return new MinState(returnType);
+  }
+
+  @Override
+  public MinState iterate(BindingTuple tuple, MinState state) {
+    Expression expression = getArguments().get(0);
+    ExprValue value = expression.valueOf(tuple);
+    if (!(value.isNull() || value.isMissing())) {
+      state.isEmptyCollection = false;
+      state.min(value);
+    }
+    return state;
+  }
+
+  @Override
+  public String toString() {
+    return String.format("min(%s)", format(getArguments()));
+  }
+
+  protected static class MinState implements AggregationState {
+    private ExprValue minResult;
+    private boolean isEmptyCollection;
+
+    MinState(ExprCoreType type) {
+      minResult = isNumber(type) ? doubleValue(Double.MAX_VALUE) : LITERAL_NULL;
+      isEmptyCollection = true;
+    }
+
+    public void min(ExprValue value) {
+      minResult = minResult.isNull() ? value : minResult.compareTo(value) < 0 ? minResult : value;
+    }
+
+    @Override
+    public ExprValue result() {
+      return isEmptyCollection ? ExprNullValue.of() : minResult;
```

Comment:
So `isEmptyCollection` is true only when `minResult` is not updated initially right? Could we remove this flag and just return `minResult`?

Reply:
Done, thanks!